tests/unit/test_check_operators/test_string_comparison.py

```diff
@@ -703,3 +703,133 @@ def test_not_matches_regex(data, comparator, dataset_type, expected_result):
         {"target": "target", "comparator": comparator}
     )
     assert result.equals(df.convert_to_series(expected_result))
+
+
+@pytest.mark.parametrize(
+    "data,separator,dataset_type,expected_result",
+    [
+        # PandasDataset tests with "/" separator
+        # Combines SENDIG examples, invalid cases, edge cases, and multiple separators
+        (
+            {
+                "target": [
```

Comment:
instead of defining a dictionary here like {"target": [1,2,3]}, you could just define the inner array like [1,2,3] and build the target below like:
df = dataset_type.from_dict({"target": data})
It just simplifies the test cases a bit. 

Same for other locations